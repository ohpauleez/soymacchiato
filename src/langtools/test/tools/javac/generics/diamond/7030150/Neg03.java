/*
 * @test /nodynamiccopyright/
 * @bug 7030150
 * @summary Type inference for generic instance creation failed for formal type parameter
 *          check that explicit type-argument that does not conform to bound is rejected
 * @compile/fail/ref=Neg03.out -XDrawDiagnostics Neg03.java
 */

class Neg03 {

    static class Foo<X> {
        <T extends Integer> Foo(T t) {}
    }

    Foo<Integer> fi1 = new <String> Foo<>(1);
    Foo<Integer> fi2 = new <String> Foo<Integer>(1);
}
