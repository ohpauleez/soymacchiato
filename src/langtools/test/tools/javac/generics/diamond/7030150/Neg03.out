Neg03.java:15:24: compiler.err.cant.apply.diamond.1: (compiler.misc.diamond: Neg03.Foo), (compiler.misc.explicit.param.do.not.conform.to.bounds: java.lang.String, java.lang.Integer)
Neg03.java:16:24: compiler.err.cant.apply.symbol.1: kindname.constructor, Foo, T, int, kindname.class, Neg03.Foo<X>, (compiler.misc.explicit.param.do.not.conform.to.bounds: java.lang.String, java.lang.Integer)
2 errors
